docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md

```diff
@@ -0,0 +1,176 @@
+
+This documentation addresses the requirement for extending the WooCommerce checkout block to accept additional data from a new inner block.
+
+## Overview
+
+Developers can extend the checkout block to add new field blocks and process additional data through the checkout POST request. This involves leveraging the extensibility interfaces provided by WooCommerce Blocks, as demonstrated in [our tutorial](https://developer.woocommerce.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/) for adding a "Not at home" shipping option.
+
+## Prerequisites
+
+- Basic understanding of React and the Gutenberg block editor.
+- Familiarity with WooCommerce Blocks' extensibility interfaces and the Store API.
+
+## Step-by-Step Guide
+
+### 1. Set Up Your Development Environment
+
+Ensure you have the following files in your project:
+
+- `index.js`: Entry point for Webpack, imports, and registers the block type.
+- `edit.js`: Handles the rendering of the block in the editor interface.
+- `block.json`: Provides metadata and configurations for the block.
+- `block.js`: Manages the block's state and user interactions.
+- `frontend.js`: Registers the checkout block component for the frontend.
+
+Refer to [this tutorial](https://developer.woocommerce.com/2023/08/07/extending-the-woocommerce-checkout-block-to-add-custom-shipping-options/) for an example of adding a custom shipping option to the checkout block.
+
+### 2. Add a new field block to the Checkout Block
+
+To add a field block to the Checkout Block you will need to add the following entries to the `block.json` file of your block:
+
+```json
+"parent": [ "woocommerce/checkout-shipping-methods-block" ],
+"attributes": {
+	"lock": {
+		"type": "object",
+		"default": {
+			"remove": true,
+			"move": true
+		}
+	}
+}
+```
+
+- The [lock attribute](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-templates/#individual-block-locking) is an object that controls whether the block can be removed or moved. By default, the lock attribute is set to allow the block to be removed and moved. However, by modifying the lock attribute, you can “force” the block to be non-removable. For example, you can set both remove and move properties to false in order to prevent the block from being removed or moved.
+- The [parent attribute](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#parent) specifies the parent block that this block should be nested within it. It determines where the block will render. In our example, the block is a child of the `woocommerce/checkout-shipping-methods-block`. This means that the your block will be rendered within the `woocommerce/checkout-shipping-methods-block`.
+
+### 3. Setting custom checkout data
+
+We can set the added field data to send it to the `wc/store/checkout` endpoint when processing orders using the function `setExtensionData`:
```

Comment:
Let's go into more detail here about what `setExtensionData` is doing, and how we get it.

It's passed to inner blocks via props.

It updates the `extensionData` key of the `wc/store/checkout` _data store_.

This key is passed as part of the request body when POSTing to the checkout endpoint.

Should we show an example of getting this from the block's props? (The example is in the tutorial)

Reply:
Maybe we can also add screenshots of Redux Dev tool showing the data store before and after the `setExtensionData` call?